refactor(YouTube): Standardize string formatting

ILoveOpenSourceApplications opened the pull request.

ILoveOpenSourceApplications committed:
refactor(YouTube): Standardize string formatting

ILoveOpenSourceApplications commented:
Regarding `Hide recommended videos`, does the patch hide anything other than "People also watched and You might also like" videos?  Because it's only hiding the labels for those and not the actual videos when using `endorsement_header_footer.eml`.

ILoveOpenSourceApplications commented:
Also, do we need two options to hide `Use this sound` button? Can't it be just bundled in the existing button itself?

ILoveOpenSourceApplications commented:
The "Hide floating button" in feed which clicked on basically opens a survey which can be used to "Tune your feed". So, that can also be chucked with "Hide surveys" button in my opinion.

inotia00 commented:
> Regarding `Hide recommended videos`, does the patch hide anything other than "People also watched and You might also like" videos? Because it's only hiding the labels for those and not the actual videos when using `endorsement_header_footer.eml`.

The filter will first try to hide the videos containing the labels. If that fails, it will only hide the labels.

> Also, do we need two options to hide `Use this sound` button? Can't it be just bundled in the existing button itself?

Shorts > Shorts player > Suggested actions > Hide Use this sound button
: Hide `Use this sound` in the Shorts player.

Shorts > Hide floating button
: Hide `Use this sound` in feed.

(When you click the `Short remixing this video` button or when you click the Shorts button in YouTube Music)

Since the settings belong to different categories, they are not integrated.

> The "Hide floating button" in feed which clicked on basically opens a survey which can be used to "Tune your feed". So, that can also be chucked with "Hide surveys" button in my opinion.

Feed > Hide floating button
: Hides buttons like [Play something](https://9to5google.com/2024/12/26/youtube-play-something-fab/) that appear in the home feed.

Integrating into `Hide surveys` doesn't seem appropriate.

inotia00 commented on extensions/shared/src/main/java/app/revanced/extension/youtube/patches/components/FeedComponentsFilter.java:
```diff
@@ -82,18 +82,11 @@ public FeedComponentsFilter() {
-                Settings.HIDE_FEED_SURVEY,
-                "selectable_item.eml",
-                "cell_button.eml"
-        );
```
The reason these filters were not merged is that one is filtered by `identifier` and the other is filtered by `path`.

Before PR:
Filtered if the litho path starts with `selectable_item.eml` and `cell_button.eml`
Filtered if the litho path contains `feed_nudge` and `_survey`.

After PR:
Filtered if the litho path contains `selectable_item.eml`, `cell_button.eml`, `feed_nudge` and `_survey`.

If YouTube has a component with a path called `video_cell_button.eml` (which is not the real component), it was not hidden before the PR, but it will be hidden after the PR.

If you are going to merge these into one filter, you should do enough testing to make sure that the merged filter does not hide unintended layouts.

ILoveOpenSourceApplications replied:
I didn't notice that, will  revert that change.

ILoveOpenSourceApplications commented:
> The filter will first try to hide the videos containing the labels. If that fails, it will only hide the labels.

Have you tested whether it hides the video? Because when I'm turning the setting on, it simply hides the label.


> Shorts > Shorts player > Suggested actions > Hide Use this sound button : Hide `Use this sound` in the Shorts player.
> 
> Shorts > Hide floating button : Hide `Use this sound` in feed.
> 
> (When you click the `Short remixing this video` button or when you click the Shorts button in YouTube Music)
> 
> Since the settings belong to different categories, they are not integrated.

I understand these settings appear in different places, but they all hide the same `Use this sound` button, which does the same thing regardless of context. Having separate toggles for each feels redundant.

It would make more sense to have a single setting that disables this button everywhere. This would simplify the experience, reduce confusion, and ensure consistent behavior across Shorts, the feed, and YouTube Music — unless there’s a specific technical reason to keep them separate.

inotia00 commented:
> Have you tested whether it hides the video? Because when I'm turning the setting on, it simply hides the label.

Here are the results from my testing:

- Only the `endorsement_header_footer.eml` label is hidden, not the videos that contain `endorsement_header_footer.eml`.
- Sometimes Members only videos that appear in the feed are hidden (but not always, so I won't explain this in the description).
- Standalone Shorts that aren't included in the Shorts shelf are hidden in search results (usually videos that aren't relevant to the search results).

> I understand these settings appear in different places, but they all hide the same `Use this sound` button, which does the same thing regardless of context. Having separate toggles for each feels redundant.
> 
> It would make more sense to have a single setting that disables this button everywhere. This would simplify the experience, reduce confusion, and ensure consistent behavior across Shorts, the feed, and YouTube Music — unless there’s a specific technical reason to keep them separate.

Some people may not want to hide the `Use this sound` button in the feed, but they may want to hide the `Use this sound` button in the Shorts player.

Since there is already a search bar in the settings, users can search for `Use this sound` and selectively hide or show it.

ILoveOpenSourceApplications commented:
> Here are the results from my testing:
> 
> * Only the `endorsement_header_footer.eml` label is hidden, not the videos that contain `endorsement_header_footer.eml`.

So, shouldn't that be changed to its own option like `Hide video recommendation labels`.

> * Sometimes Members only videos that appear in the feed are hidden (but not always, so I won't explain this in the description).

Understandable.

> * Standalone Shorts that aren't included in the Shorts shelf are hidden in search results (usually videos that aren't relevant to the search results).

Okay. On another note, is it possible to hide `Shorts remixing this video` section in description only? Does RVX have a way to identify that description is open?

> Some people may not want to hide the `Use this sound` button in the feed, but they may want to hide the `Use this sound` button in the Shorts player.
> 
> Since there is already a search bar in the settings, users can search for `Use this sound` and selectively hide or show it.

Okay, but does it hide anything else? Cause the description suggests it hides other buttons as well. Does it?

ILoveOpenSourceApplications commented:
Some of the settings that I have renamed internally, I'm unsure whether I changed it everywhere. How can I make sure that I have done it properly by the way?

inotia00 commented:
> So, shouldn't that be changed to its own option like `Hide video recommendation labels`.

The Litho filter for the `Hide recommended videos` setting needs to be refactored (It's pretty outdated).

I think it's better to make changes when refactoring.

> Okay. On another note, is it possible to hide `Shorts remixing this video` section in description only? Does RVX have a way to identify that description is open?

Yeah, you can check an example in the description filter:

https://github.com/inotia00/revanced-patches/blob/42cc8201f5dfcc05f785e578475c4e7e52c3ecfd/extensions/shared/src/main/java/app/revanced/extension/youtube/patches/components/DescriptionsFilter.java#L61

https://github.com/inotia00/revanced-patches/blob/42cc8201f5dfcc05f785e578475c4e7e52c3ecfd/extensions/shared/src/main/java/app/revanced/extension/youtube/patches/components/DescriptionsFilter.java#L125

> Okay, but does it hide anything else? Cause the description suggests it hides other buttons as well. Does it?

When hiding the `Use this sound` button in the Shorts player, [filter checks whether the camera icon is included](https://github.com/inotia00/revanced-patches/blob/42cc8201f5dfcc05f785e578475c4e7e52c3ecfd/extensions/shared/src/main/java/app/revanced/extension/youtube/patches/components/ShortsButtonFilter.java#L235C57-L235C115), so other buttons are not unintentionally hidden.

When hiding the `Use this sound` button in the feed, [filter does not check whether the camera icon is included](https://github.com/inotia00/revanced-patches/blob/42cc8201f5dfcc05f785e578475c4e7e52c3ecfd/extensions/shared/src/main/java/app/revanced/extension/youtube/patches/components/ShortsButtonFilter.java#L42), and since the keyword of the filter is `floating_action_button`, other floating buttons that are not intended may be hidden.

(There are no unintentionally hidden buttons right now, but I marked it this way in case there were server-side changes)

So the description before the PR was as follows:

```
    <string name="revanced_hide_shorts_floating_button_title">Hide floating button</string>
    <string name="revanced_hide_shorts_floating_button_summary_on">"Floating buttons like 'Use this sound' are hidden in the Shorts channel tab."</string>
    <string name="revanced_hide_shorts_floating_button_summary_off">"Floating buttons like 'Use this sound' are shown in the Shorts channel tab."</string>
```

The PR description is as follows, which does not include that other floating buttons may be hidden:

```
    <string name="revanced_hide_shorts_floating_button_title">Hide floating Use this sound button</string>
    <string name="revanced_hide_shorts_floating_button_summary_on">Floating Use this sound button is hidden.</string>
    <string name="revanced_hide_shorts_floating_button_summary_off">Floating Use this sound button is shown.</string>
```

inotia00 committed:
fix compile error

inotia00 committed:
fix compile error

inotia00 committed:
feat: Apply code review suggestions

inotia00 commented:
> The PR description is as follows, which does not include that other floating buttons may be hidden:

I'll revert this for now, If you have a more appropriate alternative, please reflect it in the next PR.

Thanks anyway.
